collections/_article/rendering-acyclic-graph.md

```diff
@@ -0,0 +1,279 @@
+---
+title: "GPU synchronization in Godot 4.3 is getting a major upgrade"
+excerpt: "Say goodbye to placing barriers, hello to the new acyclic graph"
+categories: ["progress-report"]
+author: Darío Banini
+image: /storage/blog/covers/todo.webp
+date: 2024-02-14 9:00:00
+---
+
+Since the introduction of Godot 4, [Rendering Device](https://docs.godotengine.org/en/stable/classes/class_renderingdevice.html) has been the backbone of the Forward+ and Mobile renderers. Making APIs that are both easy to use and flexible enough to cover all the features users want in a game engine is a very difficult task. As far as GPU synchronization is concerned, the high level of detail expected by APIs like Vulkan or D3D12 compared to its predecessors is worthy of a few blog posts of its own. This article will try to keep the topic of synchronization as brief as possible to focus on the problems solved within Godot instead.
+
+With the goal of reducing the difficulty of the development of the new renderers and allowing the GPU to parallelize work more effectively, the automatic construction of a [directed acyclic graph (DAG)](https://en.wikipedia.org/wiki/Directed_acyclic_graph) during rendering has been introduced at the lowest levels of the engine. [This change is already merged](https://github.com/godotengine/godot/pull/84976) in the `main` development branch of Godot and should be part of the **4.3 release**. The introduction of the graph will introduce both performance improvements and various bug fixes for issues that were very difficult to investigate. For example, after the graph was merged it was discovered that MSAA with SSAO will no longer cause artifacts in AMD Polaris ([#61415](https://github.com/godotengine/godot/issues/61415)), despite no effort being spent towards developing a specific fix for the problem.
```

Comment:
```suggestion
With the goal of reducing the difficulty of the development of the new renderers and allowing the GPU to parallelize work more effectively, the automatic construction of a [directed acyclic graph (DAG)](https://en.wikipedia.org/wiki/Directed_acyclic_graph) during rendering has been introduced at the lowest levels of the engine. [This change is already merged](https://github.com/godotengine/godot/pull/84976) in the `main` development branch of Godot and should be part of the **4.3 release**. The introduction of the graph will bring both performance improvements and various bug fixes for issues that were very difficult to investigate. For example, after the graph was merged it was discovered that MSAA with SSAO will no longer cause artifacts in AMD Polaris ([#61415](https://github.com/godotengine/godot/issues/61415)), despite no effort being spent towards developing a specific fix for the problem.
```
To avoid "introduction" and "introduce" in close proximity.